### PR TITLE
#350 Switch to mocks for kernel UserCreateTest

### LINKED ISSUE #350: Switch to mocks for kernel UserCreateTest
Switch class to be able use mocks instead of actual calls to Edge org

`/tests/src/Kernel/UserCreateTest.php`

### PR DESCRIPTION
**Pull request by arlina-espinoza**
Refactors the test to use the sdk connector's client, and apigee_mock_api_client.

### CODE REVIEW

File: tests/src/Kernel/UserCreateTest.php [hunk 1]
```diff
@@ -48,20 +67,45 @@ protected function setUp() {
...
     $this->installSchema('user', ['users_data']);
     $this->installEntitySchema('user');
+
+    $this->apigeeTestHelperSetup();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function tearDown() {
```
Comment (shadcn) [inline tests/src/Kernel/UserCreateTest.php:77]: Nice. 👍

File: tests/src/Kernel/UserCreateTest.php [hunk 2]
```diff
@@ -48,20 +67,45 @@ protected function setUp() {
...
-    $user = User::create([
+    $this->account = User::create([
       'mail' => $this->randomMachineName() . '@example.com',
       'name' => $this->randomMachineName(),
       'first_name' => $this->randomMachineName(64),
       'last_name' => $this->randomMachineName(64),
     ]);
 
-    $this->assertEquals(SAVED_NEW, $user->save());
+    $this->queueDeveloperResponse($this->account, 200);
```
Comment (shadcn) [inline tests/src/Kernel/UserCreateTest.php]: Shouldn't this return a 201?

**Review by shadcn — APPROVED** [review]
Added a comment. Otherwise good to merge.

File: tests/src/Kernel/UserCreateTest.php (hunk 2 above)
Comment (arlina-espinoza) [reply tests/src/Kernel/UserCreateTest.php]: Good one, changed to 201.

**Pull request merged**